darts/tests/models/forecasting/test_tsmixer.py

```diff
@@ -0,0 +1,601 @@
+                future_covariates=future_covariates,
+                val_series=ts_val,
+                val_past_covariates=past_covariates,
+                val_future_covariates=future_covariates,
+                verbose=False,
+            )
+
+            series = None if isinstance(ts_train, TimeSeries) else ts_train
+            y_hat = model.predict(
+                n=predict_n,
+                series=series,
+                past_covariates=past_covariates,
+                future_covariates=future_covariates,
+                num_samples=(100 if model.supports_probabilistic_prediction else 1),
+            )
+
+            if isinstance(y_hat, TimeSeries):
+                y_hat = y_hat.quantile_timeseries(0.5) if y_hat.n_samples > 1 else y_hat
+            else:
+                y_hat = [
+                    ts.quantile_timeseries(0.5) if ts.n_samples > 1 else ts
+                    for ts in y_hat
+                ]
+            return y_hat
+
+        def helper_test_prediction_accuracy(
+            self,
+            predict_n,
+            ts,
+            ts_train,
+            ts_val,
+            past_covariates,
+            future_covariates,
+            kwargs_tsmixer,
+        ):
+            """prediction should be almost equal to y_true. Absolute tolerance is set
+            to 0.2 to give some flexibility"""
+            absolute_tolerance = 0.2
+            y_hat = self.helper_fit_predict(
+                predict_n,
+                ts_train,
+                ts_val,
+                past_covariates,
+                future_covariates,
+                kwargs_tsmixer,
+            )
+
+            y_true = ts[y_hat.start_time() : y_hat.end_time()]
+
+            assert np.allclose(
+                y_true[1:-1].all_values(),
+                y_hat[1:-1].all_values(),
+                atol=absolute_tolerance,
+            )
+
+        def helper_test_prediction_shape(
+            self, predict_n, ts, ts_train, ts_val, future_covariates, kwargs_tsmixer
+        ):
+            """checks whether prediction has same number of variable as input series and
+            whether prediction has correct length"""
+            y_hat = self.helper_fit_predict(
+                predict_n, ts_train, ts_val, None, future_covariates, kwargs_tsmixer
+            )
+
+            y_hat_list = [y_hat] if isinstance(y_hat, TimeSeries) else y_hat
+            ts_list = [ts] if isinstance(ts, TimeSeries) else ts
+
+            for y_hat_i, ts_i in zip(y_hat_list, ts_list):
+                assert len(y_hat_i) == predict_n
+                assert y_hat_i.n_components == ts_i.n_components
+
+        def test_prediction_shape(self):
+            """checks whether prediction has same number of variable as input series and
+            whether prediction has correct length.
+            Test cases:
+                -   univariate
+                -   multivariate
+                -   multi-TS
```

Comment:
you can parametric these test cases using @pytest.mark.parametrize. In this way less code, same tests and it is more readable

Reply:
Thanks, it's refactored.